src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
It looks like you are not testing the code from this PR.

There are scores of different kinds of conversions that the compiler can handle. It does not make sense to test each kind of conversion in each context a conversion can occur. Unless you have some reason to suspect that the added code won't produce code for some conversion correctly, I will add only focused tests.